tests/test_cms_menus.py

```diff
@@ -76,32 +82,46 @@ def test_get_nodes(self):
             attr={"link_target": child2.link_target},
         )
 
-    def test_get_roots_with_multiple_menucontents(self):
-        """Test to check get_roots while creating two menu contents"""
-        menucontent_1 = factories.MenuVersionFactory()
-        menucontent_2 = factories.MenuVersionFactory()
-        roots = self.menu.get_roots(self.request)
-        self.assertEqual(roots.count(), 2)
-        self.assertListEqual(
-            list(roots), [menucontent_1.content.root, menucontent_2.content.root]
-        )
-
-    def test_get_roots_with_versions(self):
+    def test_get_roots_when_draft_mode_isnt_active(self):
         """This test to check versioning would group all the versions
         of menu content and return latest of all distinct menu content
         """
         menucontent_1_v1 = factories.MenuVersionFactory(state=ARCHIVED)
+        # draft version
         menucontent_1_v2 = factories.MenuVersionFactory(
             content__menu=menucontent_1_v1.content.menu
```

Comment:
As below, I think you're going to need a published version created here as well and perhaps also an unpublished one.